src/main/java/org/jitsi/jigasi/xmpp/CallControlMucActivator.java

```diff
@@ -0,0 +1,422 @@
+     */
+    private void initializeNewProvider(ProtocolProviderService pps)
+    {
+        if (!ProtocolNames.JABBER.equals(pps.getProtocolName())
+            || (pps.getAccountID() instanceof JabberAccountID
+                    && ((JabberAccountID)pps.getAccountID())
+                        .isAnonymousAuthUsed()))
+        {
+            // we do not care for anonymous logins, these are normally the xmpp
+            // sessions in a call
+            // while the call control xmpp accounts are authorized
+            return;
+        }
+
+        pps.addRegistrationStateChangeListener(this);
+
+        ProtocolProviderFactory xmppProviderFactory
+            = ProtocolProviderFactory.getProtocolProviderFactory(
+                JigasiBundleActivator.osgiContext,
+                ProtocolNames.JABBER);
+
+        new RegisterThread(
+            pps, xmppProviderFactory.loadPassword(pps.getAccountID()))
+                .start();
+    }
+
+    /**
+     * When a xmpp provder is registered to xmpp server we join it to the
+     * common room.
+     *
+     * @param evt the registration event
+     */
+    @Override
+    public void registrationStateChanged(RegistrationStateChangeEvent evt)
+    {
+        if (evt.getNewState() == RegistrationState.REGISTERED)
+        {
+            joinCommonRoom(evt.getProvider());
+        }
+    }
+
+    /**
+     * Joins the common control room.
+     * @param pps the provider to join.
+     */
+    private void joinCommonRoom(ProtocolProviderService pps)
+    {
+        OperationSetMultiUserChat muc
+            = pps.getOperationSet(OperationSetMultiUserChat.class);
+
+        String roomName = pps.getAccountID()
+            .getAccountPropertyString(ROOM_NAME_ACCOUNT_PROP);
+
+        if (roomName == null)
+        {
+            logger.warn("No brewery name specified for:" + pps);
+            return;
+        }
+
+        try
+        {
+            logger.info(
+                "Joining call control room: " + roomName + " pps:" + pps);
+
+            ChatRoom mucRoom = muc.findRoom(roomName);
+
+            mucRoom.join();
+
+            // sends initial stats, used some kind of advertising
+            // so jicofo can recognize us as real jigasi and load balance us
+            updatePresenceStatusForXmppProvider(pps);
+        }
+        catch (Exception e)
+        {
+            logger.error(e, e);
+        }
+
+        // getting direct access to the xmpp connection in order to add
+        // a listener for incoming iqs
+        if (pps instanceof ProtocolProviderServiceJabberImpl)
+        {
+            ((ProtocolProviderServiceJabberImpl) pps).getConnection()
+                .addPacketListener(this, this);
+        }
+    }
+
+    @Override
+    public void onJvbRoomJoined(GatewaySession source)
+    {
+        updatePresenceStatusForXmppProviders();
+    }
+
+    @Override
+    public void onSessionAdded(GatewaySession session)
+    {
+        session.setListener(this);
```

Comment:
Yep, cause we haven't joined the jvb room, so we can take the participants count. Will add a comment there.